### PR TITLE
Removing user java api in user connect related fragments

### PR DESCRIPTION
**Pull request by yeryomenkom**
Remove User api in:

Rewrite SendConnectRequesetFragment
Rewrite PendingConnectRequestFragment
Rewrite PendingConnectRequestManagerFragment
Rewrite SecondPageFragment
#### APK
[Download build #10508](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10508/artifact/build/artifact/wire-dev-PR1462-10508.apk)
[Download build #10509](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10509/artifact/build/artifact/wire-dev-PR1462-10509.apk)
[Download build #10520](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10520/artifact/build/artifact/wire-dev-PR1462-10520.apk)
[Download build #10522](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10522/artifact/build/artifact/wire-dev-PR1462-10522.apk)
[Download build #10525](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10525/artifact/build/artifact/wire-dev-PR1462-10525.apk)

https://github.com/wireapp/android-project/issues/16
[Download build #10531](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10531/artifact/build/artifact/wire-dev-PR1462-10531.apk)
[Download build #10534](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10534/artifact/build/artifact/wire-dev-PR1462-10534.apk)
[Download build #10535](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10535/artifact/build/artifact/wire-dev-PR1462-10535.apk)
[Download build #10536](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10536/artifact/build/artifact/wire-dev-PR1462-10536.apk)
[Download build #10538](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10538/artifact/build/artifact/wire-dev-PR1462-10538.apk)
[Download build #10540](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10540/artifact/build/artifact/wire-dev-PR1462-10540.apk)
[Download build #10548](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10548/artifact/build/artifact/wire-dev-PR1462-10548.apk)
[Download build #10550](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10550/artifact/build/artifact/wire-dev-PR1462-10550.apk)
[Download build #10555](http://192.168.120.33:8080/job/Pull%20Request%20Builder/10555/artifact/build/artifact/wire-dev-PR1462-10555.apk)

### CODE REVIEW

**Comment by yeryomenkom** [comment]
I will remove commented code and old java fragments after QA.

File: app/src/main/scala/com/waz/zclient/common/controllers/UserAccountsController.scala [hunk 1]
```diff
@@ -67,6 +67,25 @@ class UserAccountsController(implicit injector: Injector, context: Context, ec:
 
   teamMembersSignal.onUi(members => _teamMembers = members)
 
+  def getTeamIdSignal(convId: ConvId): Signal[Option[TeamId]] =
+    for {
+      zmsValue <- zms
+      conversation = zmsValue.convsStorage.conversations.find(_.id == convId)
+    } yield conversation.flatMap(_.team)
+
+  def hasRemoveConversationMemberPermission(convId: ConvId): Signal[Boolean] = {
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/common/controllers/UserAccountsController.scala]: Please merge it into one for-comprehension. If you have a signal which gives back `Option[T]` and you are sure it will be `Some[T]`, you can use a shorthand notation:
```
Some(t) <- signalOfOptionT
```
Otherwise you can use if/else:
```
someT <- signalOfOptionT
result <- if (someT.isDefined) someSignal else someOtherSignal // (eg. Signal.const(true))
```
Also, since the teamId signal is used only here, please inline it. So:
```
maybeTeamId <- convCtrl.blablabla
```

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+  override def onCreateAnimation(transit: Int, enter: Boolean, nextAnim: Int): Animation = {
+    var animation: Animation = super.onCreateAnimation(transit, enter, nextAnim)
+    val popoverLaunchMode = getControllerFactory.getConversationScreenController.getPopoverLaunchMode
+    if (popoverLaunchMode != DialogLaunchMode.AVATAR && popoverLaunchMode != DialogLaunchMode.COMMON_USER) { // No animation when request is shown in conversation list
+      val userRequester = IConnectStore.UserRequester.valueOf(getArguments.getString(ArgUserRequester))
+      if (userRequester != IConnectStore.UserRequester.CONVERSATION) {
+        val centerX: Int = ContextUtils.getOrientationIndependentDisplayWidth(getActivity) / 2
+        val centerY: Int = ContextUtils.getOrientationIndependentDisplayHeight(getActivity) / 2
+        var duration: Int = 0
+        var delay: Int = 0
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala]: Try to remove `var`s.

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  override def onViewCreated(view: View, savedInstanceState: Bundle): Unit = {
+    if (savedInstanceState != null) {
+      isShowingFooterMenu = savedInstanceState.getBoolean(StateIsShowingFooterMenu)
+    } else {
+      isShowingFooterMenu = false
+    }
+
+    userHandleView
+
+    unblockButton.setIsFilled(false)
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala]: Use `.foreach` instead of implicits, and in fact it should be enough to set the visibility in xml.

Check if the unblockButton is used at all.

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+import com.waz.zclient.pages.main.connect.UserProfileContainer
+import com.waz.zclient.participants.OptionsMenuFragment
+import com.waz.zclient.utils.ViewUtils
+import com.waz.zclient.{FragmentHelper, OnBackPressedListener, R}
+
+/**
+  * Created by admin on 3/8/18.
+  */
+
+object PendingConnectRequestManagerFragment {
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala]: Please put the `object` under the `class`.

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        import OptionsMenuFragment._
+        getChildFragmentManager
+          .beginTransaction
+          .add(R.id.fl__pending_connect_request__settings_box, newInstance(inConvList = false), Tag)
+          .commit
+      }
+    }
+  }
+
+  override def dismissUserProfile(): Unit = {
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala]: Don't use `{...}` if there is only one line of code in the method.

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    getContainer.dismissUserProfile()
+  }
+
+  override def dismissSingleUserProfile(): Unit = {
+    if (getChildFragmentManager.popBackStackImmediate) restoreCurrentPageAfterClosingOverlay()
+  }
+
+  override def showRemoveConfirmation(userId: UserId): Unit = {
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala]: Same here.

File: app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+  private lazy val userRequester = IConnectStore.UserRequester.valueOf(getArguments.getString(ArgumentUserRequester))
+
+  private lazy val usersController = inject[UsersController]
+  private lazy val userAccountsController = inject[UserAccountsController]
+  private lazy val conversationController = inject[ConversationController]
+  private lazy val zms = inject[Signal[ZMessaging]]
+
+  private lazy val accentColor = inject[AccentColorController].accentColor
+  private lazy val userToConnect = usersController.user(userToConnectId)
+  private lazy val userDisplayName = userToConnect.map(user => { Timber.d("We have user: " + user); user.getDisplayName })
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala]: If you need a log message, use ZLog

**Comment by yeryomenkom** [comment]
build again

**Comment by someoneEsle** [comment]
currently waiting this to be rebased, ping me once it's ready

File: build.gradle [hunk 8]
```diff
@@ -81,7 +81,7 @@ ext {
     // Release version number must be like this X.0(.Y)
     zMessagingReleaseVersion = System.getenv("ZMESSAGING_VERSION") ?: '120.3.442@aar'
 
-    avsVersion = '4.0.128@aar'
+    avsVersion = '4.1.16@aar'
```
Comment (deanrobertcook) [inline build.gradle]: please revert this change before merging

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+import com.waz.zclient.messages.controllers.NavigationController
+import com.waz.zclient.pages.BaseFragment
+import com.waz.zclient.pages.main.connect.UserProfileContainer
+import com.waz.zclient.participants.OptionsMenuFragment
+import com.waz.zclient.utils.ViewUtils
+import com.waz.zclient.{FragmentHelper, OnBackPressedListener, R}
+
+/**
+  * Created by admin on 3/8/18.
+  */
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala]: You can change your intellij settings to avoid putting this on every new file

File: app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+      val callback: FooterMenuCallback = new FooterMenuCallback() {
+        override def onLeftActionClicked(): Unit = {
+          showConnectButtonInsteadOfFooterMenu()
+        }
+        override def onRightActionClicked(): Unit = {
+          if (removeConvMemberFeatureEnabled)
+            getContainer.showRemoveConfirmation(userToConnectId)
+        }
+      }
+      vh.foreach(_.setCallback(callback))
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala]: ah, so the unfortunate (and slightly confusing) thing with these `ViewHolder`s is that you can *only* listen to signals in the `returning` block. If the view gets re-created, then this `returning` block will not get called again, and so that callback will not be set on the new view

File: app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+        maybeConversation <- zms.head.flatMap(_.connection.connectToUser(userToConnectId, message, uToConnect.displayName))
+      } yield maybeConversation.foreach { _ =>
+        KeyboardUtils.hideKeyboard(getActivity)
+        getContainer.onConnectRequestWasSentToUser()
+      }
+    }
+  }
+
+  override def onStop(): Unit = {
+    KeyboardUtils.hideKeyboard(getActivity)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala]: Same here

File: app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+    connectButton.onClick { _ =>
+      for {
+        uSelf <- usersController.selfUser.head
+        uToConnect <- userToConnect.head
+        message = getString(R.string.connect__message, uToConnect.name, uSelf.name)
+        maybeConversation <- zms.head.flatMap(_.connection.connectToUser(userToConnectId, message, uToConnect.displayName))
+      } yield maybeConversation.foreach { _ =>
+        KeyboardUtils.hideKeyboard(getActivity)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala]: we should move to using the `KeyboardController` instead - it's a bit more reliable

File: app/src/main/scala/com/waz/zclient/conversationpager/SecondPageFragment.scala [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+  with OnBackPressedListener
+  with ConversationManagerFragment.Container
+  with PagerControllerObserver
+  with PendingConnectRequestManagerFragment.Container
+  with ConnectRequestFragment.Container {
+
+  import Threading.Implicits.Ui
+  import SecondPageFragment._
+
+  private var currentPage: Page = _
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/conversationpager/SecondPageFragment.scala]: use Option here

Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/conversationpager/SecondPageFragment.scala]: use Option here

File: app/src/main/scala/com/waz/zclient/conversationpager/SecondPageFragment.scala [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+  }
+
+  override def onActivityResult(requestCode: Int, resultCode: Int, data: Intent): Unit = {
+    super.onActivityResult(requestCode, resultCode, data)
+    val fragment = getChildFragmentManager.findFragmentById(R.id.fl__second_page_container)
+    if (fragment != null) fragment.onActivityResult(requestCode, resultCode, data)
+  }
+
+  private def openPage(page: Page, userId: UserId) = {
+    info(s"openPage ${page.name} userId $userId")
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/conversationpager/SecondPageFragment.scala]: if `page` is null, we'll get an NPE here - so set `currentPage` first and then print

File: app/src/main/scala/com/waz/zclient/conversationpager/SecondPageFragment.scala [hunk 15]
```diff
@@ -0,0 +1,199 @@
...
+object SecondPageFragment {
+  val Tag: String = classOf[SecondPageFragment].getName
+  val ArgConversationId = "ARGUMENT_CONVERSATION_ID"
+  private val StateSecondPagePosition = "SECOND_PAGE_POSITION"
+
+
+  def newInstance = new SecondPageFragment
+
+  trait Container {
+    def onOpenUrl(url: String): Unit
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/conversationpager/SecondPageFragment.scala]: The implementations of this method should be able to use `inject[BrowserController].onOpenUrl(url)` and so you should then be able to remove this method. Once you do, the Fragment no longer needs to extend `BaseFragment`

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala [hunk 16]
```diff
@@ -0,0 +1,212 @@
...
+    userConnection.map {
+      case ConnectionStatus.PENDING_FROM_OTHER if userRequester == UserRequester.PARTICIPANTS =>
+        getString(R.string.glyph__minus)
+      case _ => ""
+    }.onUi { text => vh.foreach(_.setRightActionText(text)) }
+
+    isIgnoredConnection.map {
+      case true => getString(R.string.glyph__plus)
+      case false => getString(R.string.glyph__undo)
+    }.onUi { text => vh.foreach(_.setLeftActionText(text)) }
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala]: Looks good! You can even go one step further to reducing duplication and have:
```
 isIgnoredConnection.map {
      case true => R.string.glyph__plus
      case false => R.string.glyph__undo
    }
.map(getString)
.onUi { text => vh.foreach(_.setLeftActionText(text)) }
```

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala [hunk 17]
```diff
@@ -0,0 +1,212 @@
...
+      new ProfileAnimation(enter, duration, delay, centerX, centerY)
+    }
+  }
+
+  private def connectToUser(userId: UserId): Future[Option[ConversationData]] =
+    for {
+      uSelf <- usersController.selfUser.head
+      uToConnect <- userToConnect.head
+      message = getString(R.string.connect__message, uToConnect.name, uSelf.name)
+      maybeConversation <- zms.head.flatMap(_.connection.connectToUser(userId, message, uToConnect.displayName))
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala]: you don't have to call things `maybeXYZ`, we're pretty used to things being `Option`, and intelliJ will tell us if it is, something like `conv` is just fine :)

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala [hunk 18]
```diff
@@ -0,0 +1,212 @@
...
+      case false => getString(R.string.connect_request__cancel_request__label)
+    }.onUi { text => vh.foreach(_.setLeftActionLabelText(text)) }
+  }
+
+  private lazy val imageViewProfile = view[ImageView](R.id.pending_connect)
+
+  override def onCreateAnimation(transit: Int, enter: Boolean, nextAnim: Int): Animation = {
+    def defaultAnimation = super.onCreateAnimation(transit, enter, nextAnim)
+    def shownInConvList = {
+      val launchMode = getControllerFactory.getConversationScreenController.getPopoverLaunchMode
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala]: So, it looks as if `setPopoverLaunchedMode` is only called in one place, and it only ever has the value `DialogLaunchMode.SEARCH` or `null`. 

I guess what this is trying to tell us is whether the fragment is opened from the search or from anywhere else. When opening the fragments, we could just pass a `Boolean` flag in (`fromSearchUi`) and then behave accordingly

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala [hunk 19]
```diff
@@ -0,0 +1,212 @@
...
+    )
+    imageViewProfile.foreach(_.setImageDrawable(assetDrawable))
+
+    val backgroundContainer = findById[View](R.id.ll__pending_connect__background_container)
+    val popoverLaunchMode = getControllerFactory.getConversationScreenController.getPopoverLaunchMode
+    if (popoverLaunchMode == DialogLaunchMode.AVATAR || popoverLaunchMode == DialogLaunchMode.COMMON_USER) {
+      backgroundContainer.setClickable(true)
+    } else {
+      backgroundContainer.setBackgroundColor(Color.TRANSPARENT)
+    }
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala]: you can do
```
returning(findById[View](R.id.ll__pending_connect__background_container)) { bg => 
  ... 
}
```

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala (hunk 9 above)
Comment (deanrobertcook) [reply app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala]: please delete it too

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala (hunk 18 above)
Comment (deanrobertcook) [reply app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestFragment.scala]: We also have this `ArgUserRequester` in a lot of places. Is that representing the same information?

File: app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala [hunk 20]
```diff
@@ -0,0 +1,133 @@
...
+      else
+        Page.PENDING_CONNECT_REQUEST
+
+    navigationController.navController.setRightPage(targetLeftPage, Tag)
+  }
+
+  override def onAcceptedConnectRequest(userId: UserId): Unit =
+    getContainer.onAcceptedConnectRequest(userId)
+
+  override def onBackPressed() = false
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/PendingConnectRequestManagerFragment.scala]: Default implementation already returns false - can delete

File: app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala [hunk 21]
```diff
@@ -0,0 +1,212 @@
...
+  private lazy val userToConnect = usersController.user(userToConnectId)
+  private lazy val userDisplayName = userToConnect.map(_.getDisplayName)
+  private lazy val userName = userToConnect.map(user => StringUtils.formatHandle(user.handle.map(_.string).getOrElse("")))
+  private lazy val userPicture: Signal[ImageSource] = userToConnect.map(_.picture).collect { case Some(p) => WireImage(p) }
+  private lazy val removeConvMemberFeatureEnabled = for {
+    convId <- conversationController.currentConvId
+    permission <- userAccountsController.hasRemoveConversationMemberPermission(convId)
+  } yield permission && userRequester == IConnectStore.UserRequester.PARTICIPANTS
+
+  private lazy val footerMenuRightActionText = removeConvMemberFeatureEnabled.map {
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala]: you can inline this signal

File: app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala [hunk 22]
```diff
@@ -0,0 +1,212 @@
...
+  private lazy val usersController = inject[UsersController]
+  private lazy val userAccountsController = inject[UserAccountsController]
+  private lazy val conversationController = inject[ConversationController]
+  private lazy val keyboardController = inject[KeyboardController]
+  private lazy val zms = inject[Signal[ZMessaging]]
+
+  private lazy val accentColor = inject[AccentColorController].accentColor
+  private lazy val userToConnect = usersController.user(userToConnectId)
+  private lazy val userDisplayName = userToConnect.map(_.getDisplayName)
+  private lazy val userName = userToConnect.map(user => StringUtils.formatHandle(user.handle.map(_.string).getOrElse("")))
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala]: Some of these signals can be inlined as well

File: app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala [hunk 23]
```diff
@@ -0,0 +1,212 @@
...
+
+    footerMenu.foreach(_.setCallback(new FooterMenuCallback {
+      override def onLeftActionClicked(): Unit = showConnectButtonInsteadOfFooterMenu()
+      override def onRightActionClicked(): Unit = removeConvMemberFeatureEnabled.head foreach {
+        case true => getContainer.showRemoveConfirmation(userToConnectId)
+        case _ =>
+      }
+    }))
+
+    connectButton.onClick { _ =>
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/connect/SendConnectRequestFragment.scala]: I forgot to mention, if you look at this implementation of `onClick`, you'll see it's actually on the `ViewHolder` itself - this allows us to set the `onClick` in the `returning` block of the viewHolder's definition

File: app/src/main/scala/com/waz/zclient/conversation/ParticipantDetailsTab.scala [hunk 24]
```diff
@@ -100,9 +100,10 @@ class ParticipantDetailsTab(val context: Context, callback: FooterMenuCallback)
   (for {
     convId     <- participantsController.conv.map(_.id)
     groupOrBot <- participantsController.isGroupOrBot
-  } yield (groupOrBot, convId)).map {
-    case (false, _)     if userAccountsController.hasCreateConversationPermission               => R.string.glyph__more
-    case (true, convId) if userAccountsController.hasRemoveConversationMemberPermission(convId) => R.string.glyph__minus
+    hasRemoveMeberPermission <- userAccountsController.hasRemoveConversationMemberPermission(convId)
+  } yield (groupOrBot, convId, hasRemoveMeberPermission)).map {
+    case (false, _, _)  if userAccountsController.hasCreateConversationPermission => R.string.glyph__more
+    case (true, convId, true) => R.string.glyph__minus
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/conversation/ParticipantDetailsTab.scala]: `convId` is not used here - you can remove it from the tuple

File: app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala [hunk 25]
```diff
@@ -404,15 +406,19 @@ class ConversationListManagerFragment extends Fragment
...
-  override def onAcceptedPendingOutgoingConnectRequest(conversation: ConvId) = {
-    verbose(s"onAcceptedPendingOutgoingConnectRequest: $conversation")
-    convController.selectConv(conversation, ConversationChangeRequester.CONNECT_REQUEST_ACCEPTED)
-  }
+//  override def onAcceptedPendingOutgoingConnectRequest(userId: UserId) = {
+//    verbose(s"onAcceptedPendingOutgoingConnectRequest: $userId")
+//    userAccountsController.getConversationId(userId).flatMap { convId =>
+//      convController.selectConv(convId, ConversationChangeRequester.CONNECT_REQUEST_ACCEPTED)
+//    }
+//  }
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/conversationlist/ConversationListManagerFragment.scala]: Can delete this block

File: app/src/main/scala/com/waz/zclient/conversationpager/SecondPageFragment.scala [hunk 26]
```diff
@@ -0,0 +1,176 @@
...
+      else if (switchingToPendingConnectRequest) openPage(Page.CONNECT_REQUEST_PENDING, userId)
+      else openPage(Page.MESSAGE_STREAM, userId)
+    }
+
+  }
+
+  override def onActivityResult(requestCode: Int, resultCode: Int, data: Intent): Unit = {
+    super.onActivityResult(requestCode, resultCode, data)
+    val fragment = getChildFragmentManager.findFragmentById(R.id.fl__second_page_container)
+    if (fragment != null) fragment.onActivityResult(requestCode, resultCode, data)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/conversationpager/SecondPageFragment.scala]: we have a method in the `FragmentHelper` class `withFragmentOpt(id: Int)(f: Option[Fragment] => Unit)`

File: app/src/main/scala/com/waz/zclient/conversationpager/SecondPageFragment.scala [hunk 27]
```diff
@@ -0,0 +1,176 @@
...
+        case (transaction, Page.CONNECT_REQUEST_INBOX | Page.CONNECT_REQUEST_PENDING) =>
+          transaction.setCustomAnimations(
+            R.anim.fragment_animation_second_page_slide_in_from_right,
+            R.anim.fragment_animation_second_page_slide_out_to_left
+          )
+      }
+      .foreach(transaction => transaction.commit())
+  }
+
+  override def onOpenUrl(url: String): Unit = browserController.openUrl(url)
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/conversationpager/SecondPageFragment.scala]: method not used - can delete the trait too
